src/platform/posix/tcp_ip_channel.c

```diff
@@ -486,6 +496,8 @@ void TcpIpChannel_ctor(TcpIpChannel *self, const char *host, unsigned short port
   self->fd = 0;
   self->state = NETWORK_CHANNEL_STATE_UNINITIALIZED;
 
+  self->super.get_dest_channel_id = TcpIpChannel_get_dest_channel_id;
+  self->super.get_connection_state = TcpIpChannel_get_connection_state;
```

Comment:
It seems the `connection_state` mapping was missing so I secretly added it here also

Reply:
Ops! Thanks!